contribs/gmf/src/directives/layertree.js

```diff
@@ -630,22 +650,85 @@ gmf.LayertreeController.prototype.getNodeState = function(treeCtrl) {
+ *
+ * @param  {Object} date The date object
+ * @return {string} ISO date string without millesceond
+ * @private
+ */
+gmf.LayertreeController.prototype.toISOdateString_ = function(date) {
+  return date.toISOString().replace(/\.\d{3}/, '');
+};
+
+
+/**
+ * LayertreeController.prototype.nodeTimeToISO_ - get the ISO-8601 string duration/time
+ * for the node using its time parameter
+ * @private
+ * @param  {GmfThemesNode} node node
+ * @return {string|undefined} ISO-8601 string duration/time
+ */
+gmf.LayertreeController.prototype.nodeTimeToISO_ = function(node) {
+  if (!node.time)
+    return undefined;
+
+  return node.time['maxValue']
+    ? this.toISOdateString_(new Date(node.time['minValue'])) + '/' + this.toISOdateString_(new Date(node.time['maxValue']))
+    : this.toISOdateString_(new Date(node.time['minValue']));
+};
+
+
+/**
+ * Update the TIME parameter of the source of the layer attached to the given
+ * layertree contoller
+ * LayertreeController.prototype.updateWMSTimeLayerState - description
+ * @param {ngeo.LayertreeController} layertreeCtrl ngeo layertree controller
+ * @param {{start : Object, end : Object}} time The start
+ * and optionally the end datetime (for time range selection) selected by user
+ * @export
+ */
+gmf.LayertreeController.prototype.updateWMSTimeLayerState = function(layertreeCtrl, time) {
+  if (time) {
+    var layer = /** @type {ol.layer.Image} */ (layertreeCtrl.layer || this.retrieveFirstParentTree_(layertreeCtrl).layer);
+    if (layer) {
+      var source = /** @type {ol.source.ImageWMS} */ (layer.getSource());
+      var timeParam = time.end
+        ? this.toISOdateString_(time.start) + '/' + this.toISOdateString_(time.end)
+        : this.toISOdateString_(time.start);
+      this.updateWMSLayerState_(layer, source.getParams()['LAYERS'], timeParam);
+    }
+  }
+};
+
+
+/**
  * Update the LAYERS parameter of the source of the given WMS layer.
  * @param {ol.layer.Image} layer The WMS layer.
- * @param {Array.<string>} names The array of names that will be used to set
+ * @param {string} names The names that will be used to set
  * the LAYERS parameter.
+ * @param {string=} opt_time The start
+ * and optionally the end datetime (for time range selection) selected by user
+ * in a ISO-8601 string datetime or time interval format
  * @private
  */
 gmf.LayertreeController.prototype.updateWMSLayerState_ = function(layer,
-    names) {
+    names, opt_time) {
   // Don't send layer without parameters, hide layer instead;
   if (names.length <= 0) {
     layer.setVisible(false);
   } else {
     layer.setVisible(true);
-    names.reverse();
+    //names.reverse();
     var source = /** @type {ol.source.ImageWMS} */ (layer.getSource());
-    source.updateParams({'LAYERS': names.join(',')});
+    var prevNames = source.getParams()['LAYERS'];
+    var prevTime = source.getParams()['TIME'];
+    if (names !== prevNames || opt_time !== prevTime) {
```

Comment:
Normally it should not happen but changes can come from different view which don't now the state of the parameters. This test is useful at initialization